apluslms_roman/cli.py

```diff
@@ -223,46 +226,93 @@ def add_cli_actions(parser):
         help=_("a list of sub commands:"))
     parser.use_subparsers(
         title=_("Commands"),
-        description=_("Top level commands. Check the per command help with `%(prog)s COMMAND --help`."))
+        description=_(
+            "Top level commands. Check the per command help " +
+            "with `%(prog)s COMMAND --help`."))
 
     build = parser.add_parser('build', aliases=['b'],
         callback=build_action,
         help=_("build the course (default action)"))
     build.add_argument('-s', '--steps', nargs='+',
         help=_("select which steps to build and in which order (use either index or step name)"))
-    build.add_argument('-l', '--list-steps',
-        action='store_true',
-        help=_("list all available steps and exit"))
 
     # build is the default callback. set defaults for it
     build.copy_defaults_to(parser)
     parser.set_callback(build_action)
 
+
+    parser.add_parser('init',
+        callback=init_action,
+        help=("create roman settings file in current directory"))
+
+
     config = parser.add_parser('config', aliases=['c'],
         callback=config_print_action,
         help=_("manage %(prog)s configuration"))
-    config.add_argument('-u', '--user', action='store_true',
+
+    config.add_argument('-g', '--global', action='store_true', dest='global_',
         help=_("limit actions to the global %(prog)s configuration"))
+
     config.add_argument('-p', '--project', action='store_true',
         help=_("limit actions to the project configuration"))
+
     with config.use_subparsers(title=_("Config actions")):
         config.add_parser('print', aliases=['p'],
             callback=config_print_action,
-            help=_("print configurations (default action)"))
+            help=_("print configurations (default action)")
+            )
+        setval = config.add_parser('set', aliases=['s'],
+            callback=config_set_action,
+            help=_("change/add values in the configuration"))
+        setval.add_argument('values', nargs='+', help=_("format: key=val"))
+
+        delval = config.add_parser('remove', aliases=['rm'],
+            callback=config_rm_action,
+            help=_("remove values from the configuration"))
+        delval.add_argument('keys', nargs='+')
+
+
+    step = parser.add_parser('step',
+        help=_("actions for editing steps in the project configuration"))
+
+    with step.use_subparsers(title=_("Step actions")):
+        step.add_parser('list', aliases=['ls'],
+            callback=step_list_action,
+            help=_("list all steps and exit"))
+
+        add = step.add_parser('add', aliases=['a'],
+            callback=step_add_action,
+            help=_("add step to roman.yml"))
+        add.add_argument('img')
+        add.add_argument('-n', '--name')
+        add.add_argument('-e', '--env', nargs='+', help=_("format: key=val"))
```

Comment:
hmm.. how does `nargs='+'` or `nargs='*'` differ from `action='append'` in this kind of case? Documentation uses append for non-positional arguments and nargs for positional arguments in examples. Any case, `nargs='*'` probably should be used at least, so the `-e` wouldn't be required.

Reply:
The `nargs='+'` doesn't mean that `--env` is required, it means that if env is used, at least one argument is required, which is how we want it to work. Because `--env` has `--` in front of it, it isn't a required argument. `action='append'` means that for each `-e` there needs to be exactly one argument, so if the user wants to put several values into the env, they need to write the flag several times.

Reply:
Hahaa. Great! Then it sounds exactly the correct. Yey. :)